### ISSUE TITLE
When LWS is integrated into ESP_IFD, any make produces a new binary even of nothing has changed

### ISSUE THREAD

**Issue by JimGaluska** [warmcat/libwebsockets#1118]
I have installed Expressif's ESP-IDF and then cloned esp32 branch (d6899f3) into the components directory of the IDF.  Once I did that I noticed that cmake will cause the project binary to be rebuild, even if nothing has changed.  In fact, a new binary will be created even if the project does not even use LWS.

**Comment by lws-team** [comment]
...  I already replied to this as follows:

> I have installed Expressif ESP_IDF and then install LWS into the components directory of the ESP_IDF.

There's no reason to do that, use the same technique as -factory and the test demo and put lws in your project.

I don't understand what you mean by "new binary" either, the component should generate a .a.

If you really want to put it in esp-idf, well that's fine, but maybe you can debug the issue?

**Comment by JimGaluska** [comment]
What is happening is that the IDF's make is acting as if LWS is out of date, so it compiles everything that needs to be compiles (which is nothing) and then links all the components into a new binary.

I have moved (re-cloned) LWS to the project directory (.../project/components/libwebsockets).  While this will definitely fix the issue where it is impacting project not using LWS, it is still causing a new binary to be built.  I have attached the trace when you execute IDF's make twice in a row without changing anything else. you will see that it generates new keys, runs cmake, and creates a new application binary each time.

[make_issue.txt](https://github.com/warmcat/libwebsockets/files/1530224/make_issue.txt)
